zilliqa/src/exec.rs

```diff
@@ -316,6 +316,18 @@ impl State {
         Ok(())
     }
 
+    pub fn get_stakers_at_block(&mut self, block: &Block) -> Result<Vec<NodePublicKey>> {
+        let curr_root_hash = self.root_hash()?;
+        let block_root_hash = block.state_root_hash();
+        self.set_to_root(H256(block_root_hash.0));
+
+        let committee = self.get_stakers();
+
+        self.set_to_root(H256(curr_root_hash.0));
```

Comment:
As above, `self.at_root()` to eliminate `mut`

Reply:
FWIW I think we should only have `get_stakers_at_block` and we should remove `get_stakers` (at the implicit block that someone else set before you called this and may as well be a random value).

Reply:
I would also prefer if the state root hash manipulation happened one level up in the call stack - i.e. we manipulate the `State` in `Consensus` to set it to the right height, then call `.get_stakers()`.